packages/@uppy/aws-s3/src/index.js

```diff
@@ -122,6 +122,10 @@ export default class AwsS3 extends BasePlugin {
 
     this.opts = { ...defaultOptions, ...opts }
 
+    if ('metaFields' in opts) {
```

Comment:
Could the warning be displayed only if the new option has not been supplied? In case someone is trying to write a Uppy plugin that works on both 2.x and 3.x branch.

Reply:
Good point